File: Audio Inpainting/utils/evaluation/snr.py
import numpy as np

import numpy as np

def SNR(xRef: np.ndarray, xEst: np.ndarray) -> float:
    """
    Compute the Signal-to-Noise Ratio (SNR).

    Args:
        xRef (np.ndarray): Reference signal.
        xEst (np.ndarray): Estimate signal.

    Returns:
        float: The computed SNR.
    """
    # Signal to noise ratio (SNR)
    # Add eps to avoid NaN/Inf values
    snr = 10 * np.log10((np.sum(np.abs(xRef)**2) + np.finfo(float).eps) / (np.sum(np.abs(xRef - xEst)**2) + np.finfo(float).eps))
    return snr

File: AudioInpaintingToolbox/Utils/evaluation/SNR.m
function snr = SNR(xRef,xEst)
% Signal-to-noise Ratio
%
% Usage: snr = SNR(xRef,xEst)
%
%
% Inputs:
%          - xRef - reference signal
%          - xEst - estimate signal
%
% Outputs:
%          - snr - SNR
%
%
% -------------------
%
% Audio Inpainting toolbox
% Date: June 28, 2011
% By Valentin Emiya, Amir Adler, Maria Jafari
% This code is distributed under the terms of the GNU Public License version 3 (http://www.gnu.org/licenses/gpl.txt).
% Signal to noise ratio (SNR)

% Add eps to avoid NaN/Inf values
snr = 10*log10((sum(abs(xRef(:)).^2)+eps)/sum((abs(xRef(:)-xEst(:)).^2)+eps));

return
